### PR TITLE
Geoip

### PR DESCRIPTION
**Pull request by pplonsitetesting**
Added a Transformer for obtaining Geographic information from an IP address.

### CODE REVIEW

File: spec/optimus_prime/transformers/geoip_lookup_spec.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      }
+    ]
+  end
+
+  before do
+    stub_request(:get, api_url + "188.32.194.21")
+      .to_return(status: 200, body: response_1)
+  end
+
+  context 'lookup geoip information' do
```
Comment (Anumodana) [inline spec/optimus_prime/transformers/geoip_lookup_spec.rb]: Can you add a test case where the service returns an error code? You can see an example error [here](https://freegeoip.net/json/0.0.0.1111).

More information see https://github.com/fiorix/freegeoip/blob/8b86589c43087fce011fbeaaa4a96da374fcf022/freegeoip.go#L45-L66.

**Comment by jorinvo** [comment]
Nice! :+1: for documentation :wink:

File: lib/optimus_prime/transformers/geoip_lookup.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      end
+
+      def write(record)
+        push lookup_geoip record
+      end
+
+      private
+
+      def lookup_geoip(record)
+        resp = JSON.parse(RestClient.get(@api_url + record[@ip_field]).body)
```
Comment (TamerShlash) [inline lib/optimus_prime/transformers/geoip_lookup.rb]: If this request fails due to a network error, the whole pipeline will fail. Usually, we support a number of retries (3-5) with exponential backoff to make failure as hard as possible.

File: lib/optimus_prime/transformers/geoip_lookup.rb [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+            save_database_gz(response)
+          end
+        end
+        uncompress_database_gz
+      end
+
+      def save_database_gz(response)
+        open @gz_db_file, 'w' do |io|
+          response.read_body do |chunk|
+            print '.'
```
Comment (Anumodana) [inline lib/optimus_prime/transformers/geoip_lookup.rb:135]: Why did you add `print .` here?

Comment (pplonsitetesting) [inline lib/optimus_prime/transformers/geoip_lookup.rb:135]: This just shows the download progress of the file by printing a . for every chunk downloaded. I figured if someone was running it from a CLI it should show progress instead of appearing locked up (ex slow internet)

**Comment by iammchat** [comment]
:+1:

**Comment by Anumodana** [comment]
LGTM :+1:

**Comment by Anumodana** [comment]
@T-Dnzt What do you think?

**Comment by T-Dnzt** [comment]
:+1: Looks good!

**Comment by jorinvo** [comment]
@pplonsitetesting :+1: 

If you want to, you can delete the branch now to keep the repo clean.
